go-controller/pkg/kubevirt/router.go

```diff
@@ -243,29 +296,38 @@ func EnsureRemoteZonePodAddressesToNodeRoute(watchFactory *factory.WatchFactory,
 		}); err != nil {
 			return fmt.Errorf("failed adding static route to remote pod: %v", err)
 		}
+
 	}
 	return nil
 }
 
-func virtualMachineReady(watchFactory *factory.WatchFactory, pod *corev1.Pod) (bool, error) {
-	isMigratedSourcePodStale, err := IsMigratedSourcePodStale(watchFactory, pod)
-	if err != nil {
-		return false, err
+func reconcileClusterDefaultNetworkRemoteZoneNeighbours(watchFactory *factory.WatchFactory, nbClient libovsdbclient.Client, pod *corev1.Pod, podAnnotation *util.PodAnnotation, nadName string) error {
+	//TODO: Implement neighbour update for non IC
+	if !config.OVNKubernetesFeature.EnableInterconnect {
+		return nil
 	}
-	if util.PodWantsHostNetwork(pod) || !IsPodLiveMigratable(pod) || isMigratedSourcePodStale {
-		return false, nil
+	vmRunningAtNodeOwningSubnet, err := nodeContainsPodSubnet(watchFactory, pod.Spec.NodeName, podAnnotation, nadName)
+	if err != nil {
+		return err
 	}
 
-	// When a virtual machine start up this
-	// label is the signal from KubeVirt to notify that the VM is
-	// ready to receive traffic.
-	targetNode := pod.Labels[kubevirtv1.NodeNameLabel]
-
-	// This annotation only appears on live migration scenarios and it signals
-	// that target VM pod is ready to receive traffic so we can route
-	// taffic to it.
-	targetReadyTimestamp := pod.Annotations[kubevirtv1.MigrationTargetReadyTimestamp]
+	currentNode := os.Getenv("K8S_NODE")
```

Comment:
This seems to be the first time we introduce direct dependency on `K8S_NODE` in the code, it means with this change all deployment types need to ensure it is set.
I think an alternative might be to depend on the `zone` name but I don't like it all that much either so I am ok to keep this one as is. 
Let's just check it is set, for sanity.